tests/components/august/mocks.py

```diff
@@ -39,44 +39,97 @@ def _mock_get_config():
 
 @mock.patch("homeassistant.components.august.Api")
 @mock.patch("homeassistant.components.august.Authenticator.authenticate")
-async def _mock_setup_august(hass, api_mocks_callback, authenticate_mock, api_mock):
+async def _mock_setup_august(hass, api_instance, authenticate_mock, api_mock):
     """Set up august integration."""
     authenticate_mock.side_effect = MagicMock(
         return_value=_mock_august_authentication("original_token", 1234)
     )
-    api_mocks_callback(api_mock)
+    api_mock.return_value = api_instance
     assert await async_setup_component(hass, DOMAIN, _mock_get_config())
     await hass.async_block_till_done()
     return True
 
 
-async def _create_august_with_devices(hass, lock_details=[], doorbell_details=[]):
+async def _create_august_with_devices(
+    hass, lock_details=[], doorbell_details=[], api_call_side_effects={}
```

Comment:
Mutable objects as default arguments is generally error prone.

Is it needed?

Reply:
Refactored to take devices in a single argument